lib/setup/setupPacketManager.js

```diff
@@ -29,238 +30,186 @@ class PacketManager {
+            return stdout;
+        } catch (e) {
+            // Log error and continue
+            this.logger.error(`Cannot install ${packet}: ${e.stderr || e.stdout || e}`);
         }
-        return new Promise(resolve => {
-            try {
-                const _cmd = exec((this.sudo ? 'sudo ' : '') + this.manager + ' install ' + packet + ' -y', {windowsHide: true}, (err, stdout) => {
-                    if (err) {
-                        this.logger.error(`Cannot install ${packet}: ${err.toString()}`);
-                        resolve();
-                    } else {
-                        resolve(stdout);
-                    }
-                });
-
-                _cmd.on('error', () => resolve());
-            } catch (e) {
-                resolve();
-            }
-        });
     }
 
-    _installPackets(packets, cb) {
+    /**
+     * Installs multiple packets. The returned Promise contains the list of failed packets
+     * @param {string[]} packets
+     * @returns {Promise<string[]>}
+     */
+    async _installPackets(packets) {
         if (!packets || !packets.length) {
-            cb && cb();
+            return;
         } else {
-            const packet = packets.shift();
-
-            this._installPacket(packet)
-                .catch(e => this.logger.error(`Cannot install "${packet}": ` + e))
-                .then(() => setImmediate(() => this._installPackets(packets, cb)));
+            /** @type {string[]} */
+            const failed = [];
+            // Install all packets
+            for (const packet of packets) {
+                try {
+                    await this._installPacket(packet);
+                } catch (e) {
+                    failed.push(packet);
+                    this.logger.error(`Cannot install "${packet}": ${e}`);
+                    // Continue with the next packet
+                }
+            }
+            return failed;
         }
     }
 
-    install(packets) {
+    /**
+     * Installs all given packets
+     * @param {string[] | string} packets
+     * @returns {Promise<void>}
+     */
+    async install(packets) {
         packets = packets || [];
         if (!(packets instanceof Array)) {
             packets = [packets];
         }
 
-        return this.ready()
-            .then(() => this.checkInstalled(packets))
-            .then(result => {
-                // filter non installed packets
-                packets = packets.filter((name, i) => !result[i]);
-                // install packets one after each other
-                return new Promise(resolve =>
-                    this._installPackets(packets, () => resolve()));
-            })
-            .catch(() => this.logger && this.logger.warn(`Some ${this.manager || 'OS'} packages could not be installed. Please install them manually`));
+        await this.ready();
+        const installed = await this.checkInstalled(packets);
```

Comment:
I think we also shpuld add an info log like "The following OS packages are already installed ..." ... Or ?!

Reply:
Are you looking at an old version? The current one prints the missing packages:
```
this.logger.warn(`The following ${this.manager || 'OS'} packages could not be installed: ${failed.join(', ')}. Please install them manually.`);
```

Reply:
Yes but my idea is to add a "the folloing are installed" confirmation". Currently the package manager is completely quiet when evrything is installed ... this also mean noone  sees if it worked at all to check for them :-) Thats why the idea to post whats there already :-)

Reply:
ahhh